tests/Unit/NumericalAlgorithms/Interpolation/Test_PredictedZeroCrossing.cpp

```diff
@@ -10,6 +10,50 @@
 #include "NumericalAlgorithms/Interpolation/LinearLeastSquares.hpp"
 #include "NumericalAlgorithms/Interpolation/PredictedZeroCrossing.hpp"
 
+namespace {
+
+void test_predicted_zero_crossing_datavector() {
+  // Set up random number generator
+  MAKE_GENERATOR(gen);
+  std::uniform_real_distribution<> dist(-10., 10.);
+  std::uniform_real_distribution<> error_dist(-1e-8, 1e-8);
+
+  DataVector expected_zero_crossing_value(3, 0.);
+  for (size_t i = 0; i < expected_zero_crossing_value.size(); ++i) {
+    expected_zero_crossing_value[i] = dist(gen);
+  }
+  CAPTURE(expected_zero_crossing_value);
+  DataVector slope(3, 0.);
+  for (size_t i = 0; i < slope.size(); ++i) {
+    slope[i] = dist(gen);
+  }
+  CAPTURE(slope);
+
+  const DataVector x_values{0., 1., 2.};
+  std::vector<DataVector> y_values{};
+
+  DataVector b(3, 0.);
+  for (size_t i = 0; i < b.size(); ++i) {
+    b[i] = -slope[i] * expected_zero_crossing_value[i];
+  }
+
+  for (size_t i = 0; i < x_values.size(); ++i) {
+    double error = error_dist(gen);
+    y_values.push_back(DataVector{slope[i] * x_values[0] + b[i] + error,
+                                  slope[i] * x_values[1] + b[i] + error,
+                                  slope[i] * x_values[2] + b[i] + error});
```

Comment:
Is there a reason why each component has to have the same error? Or can each component have a different error?

Reply:
No reason besides just adding a small perturbation for each calculation, I think different would values would be better, I can change that if that's preferred. @knelli2